diff -up Idris2-0.4.0/Makefile.destdir Idris2-0.4.0/Makefile
--- Idris2-0.4.0/Makefile.destdir	2021-06-24 03:09:33.000000000 +0800
+++ Idris2-0.4.0/Makefile	2021-07-06 18:42:56.961947648 +0800
@@ -153,35 +153,35 @@ install-with-src-api: src/IdrisPaths.idr
 	${IDRIS2_BOOT} --install-with-src ${IDRIS2_LIB_IPKG}
 
 install-idris2:
-	mkdir -p ${PREFIX}/bin/
-	install ${TARGET} ${PREFIX}/bin
+	mkdir -p ${DESTDIR}${PREFIX}/bin/
+	install ${TARGET} ${DESTDIR}${PREFIX}/bin
 ifeq ($(OS), windows)
-	-install ${TARGET}.cmd ${PREFIX}/bin
+	-install ${TARGET}.cmd ${DESTDIR}${PREFIX}/bin
 endif
-	mkdir -p ${PREFIX}/lib/
-	install support/c/${IDRIS2_SUPPORT} ${PREFIX}/lib
-	mkdir -p ${PREFIX}/bin/${NAME}_app
-	install ${TARGETDIR}/${NAME}_app/* ${PREFIX}/bin/${NAME}_app
+	mkdir -p ${DESTDIR}${PREFIX}/lib/
+	install support/c/${IDRIS2_SUPPORT} ${DESTDIR}${PREFIX}/lib
+	mkdir -p ${DESTDIR}${PREFIX}/bin/${NAME}_app
+	install ${TARGETDIR}/${NAME}_app/* ${DESTDIR}${PREFIX}/bin/${NAME}_app
 
 install-support:
-	mkdir -p ${PREFIX}/${NAME_VERSION}/support/docs
-	mkdir -p ${PREFIX}/${NAME_VERSION}/support/racket
-	mkdir -p ${PREFIX}/${NAME_VERSION}/support/gambit
-	mkdir -p ${PREFIX}/${NAME_VERSION}/support/js
-	install support/docs/* ${PREFIX}/${NAME_VERSION}/support/docs
-	install support/racket/* ${PREFIX}/${NAME_VERSION}/support/racket
-	install support/gambit/* ${PREFIX}/${NAME_VERSION}/support/gambit
-	install support/js/* ${PREFIX}/${NAME_VERSION}/support/js
+	mkdir -p ${DESTDIR}${PREFIX}/${NAME_VERSION}/support/docs
+	mkdir -p ${DESTDIR}${PREFIX}/${NAME_VERSION}/support/racket
+	mkdir -p ${DESTDIR}${PREFIX}/${NAME_VERSION}/support/gambit
+	mkdir -p ${DESTDIR}${PREFIX}/${NAME_VERSION}/support/js
+	install support/docs/* ${DESTDIR}${PREFIX}/${NAME_VERSION}/support/docs
+	install support/racket/* ${DESTDIR}${PREFIX}/${NAME_VERSION}/support/racket
+	install support/gambit/* ${DESTDIR}${PREFIX}/${NAME_VERSION}/support/gambit
+	install support/js/* ${DESTDIR}${PREFIX}/${NAME_VERSION}/support/js
 	@${MAKE} -C support/c install
 	@${MAKE} -C support/refc install
 	@${MAKE} -C support/chez install
 
 install-libs:
-	${MAKE} -C libs/prelude install IDRIS2?=${TARGET} IDRIS2_PATH=${IDRIS2_BOOT_PATH}
-	${MAKE} -C libs/base install IDRIS2?=${TARGET} IDRIS2_PATH=${IDRIS2_BOOT_PATH}
-	${MAKE} -C libs/contrib install IDRIS2?=${TARGET} IDRIS2_PATH=${IDRIS2_BOOT_PATH}
-	${MAKE} -C libs/network install IDRIS2?=${TARGET} IDRIS2_PATH=${IDRIS2_BOOT_PATH}
-	${MAKE} -C libs/test  install IDRIS2?=${TARGET} IDRIS2_PATH=${IDRIS2_BOOT_PATH}
+	${MAKE} -C libs/prelude install IDRIS2?=${TARGET} IDRIS2_PATH=${IDRIS2_BOOT_PATH} IDRIS2_PREFIX=${DESTDIR}${PREFIX}
+	${MAKE} -C libs/base install IDRIS2?=${TARGET} IDRIS2_PATH=${IDRIS2_BOOT_PATH} IDRIS2_PREFIX=${DESTDIR}${PREFIX}
+	${MAKE} -C libs/contrib install IDRIS2?=${TARGET} IDRIS2_PATH=${IDRIS2_BOOT_PATH} IDRIS2_PREFIX=${DESTDIR}${PREFIX}
+	${MAKE} -C libs/network install IDRIS2?=${TARGET} IDRIS2_PATH=${IDRIS2_BOOT_PATH} IDRIS2_PREFIX=${DESTDIR}${PREFIX}
+	${MAKE} -C libs/test  install IDRIS2?=${TARGET} IDRIS2_PATH=${IDRIS2_BOOT_PATH} IDRIS2_PREFIX=${DESTDIR}${PREFIX}
 
 install-with-src-libs:
 	${MAKE} -C libs/prelude install-with-src IDRIS2?=${TARGET} IDRIS2_PATH=${IDRIS2_BOOT_PATH}
diff -up Idris2-0.4.0/support/c/Makefile.destdir Idris2-0.4.0/support/c/Makefile
--- Idris2-0.4.0/support/c/Makefile.destdir	2021-06-24 03:09:34.000000000 +0800
+++ Idris2-0.4.0/support/c/Makefile	2021-07-06 13:00:11.804301987 +0800
@@ -51,8 +51,8 @@ cleandep: clean
 .PHONY: install
 
 install: build
-	mkdir -p ${PREFIX}/idris2-${IDRIS2_VERSION}/lib
-	mkdir -p ${PREFIX}/idris2-${IDRIS2_VERSION}/include
-	install -m 755 $(DYLIBTARGET) ${PREFIX}/idris2-${IDRIS2_VERSION}/lib
-	install -m 644 $(LIBTARGET)   ${PREFIX}/idris2-${IDRIS2_VERSION}/lib
-	install -m 644 *.h            ${PREFIX}/idris2-${IDRIS2_VERSION}/include
+	mkdir -p ${DESTDIR}${PREFIX}/idris2-${IDRIS2_VERSION}/lib
+	mkdir -p ${DESTDIR}${PREFIX}/idris2-${IDRIS2_VERSION}/include
+	install -m 755 $(DYLIBTARGET) ${DESTDIR}${PREFIX}/idris2-${IDRIS2_VERSION}/lib
+	install -m 644 $(LIBTARGET)   ${DESTDIR}${PREFIX}/idris2-${IDRIS2_VERSION}/lib
+	install -m 644 *.h            ${DESTDIR}${PREFIX}/idris2-${IDRIS2_VERSION}/include
diff -up Idris2-0.4.0/support/chez/Makefile.destdir Idris2-0.4.0/support/chez/Makefile
--- Idris2-0.4.0/support/chez/Makefile.destdir	2021-06-24 03:09:34.000000000 +0800
+++ Idris2-0.4.0/support/chez/Makefile	2021-07-06 13:00:11.804301987 +0800
@@ -10,8 +10,8 @@ clean:
 build: support-sep.ss
 
 install: build
-	mkdir -p ${PREFIX}/idris2-${IDRIS2_VERSION}/support/chez
-	install *.ss ${PREFIX}/idris2-${IDRIS2_VERSION}/support/chez
+	mkdir -p ${DESTDIR}${PREFIX}/idris2-${IDRIS2_VERSION}/support/chez
+	install *.ss ${DESTDIR}${PREFIX}/idris2-${IDRIS2_VERSION}/support/chez
 
 support-sep.ss: support.ss
 	# start library header
diff -up Idris2-0.4.0/support/refc/Makefile.destdir Idris2-0.4.0/support/refc/Makefile
--- Idris2-0.4.0/support/refc/Makefile.destdir	2021-06-24 03:09:34.000000000 +0800
+++ Idris2-0.4.0/support/refc/Makefile	2021-07-06 13:00:11.804301987 +0800
@@ -45,5 +45,5 @@ cleandep: clean
 .PHONY: install
 
 install: build
-	mkdir -p ${PREFIX}/idris2-${IDRIS2_VERSION}/refc
-	install $(LIBTARGET) *.h ${PREFIX}/idris2-${IDRIS2_VERSION}/refc
+	mkdir -p ${DESTDIR}${PREFIX}/idris2-${IDRIS2_VERSION}/refc
+	install $(LIBTARGET) *.h ${DESTDIR}${PREFIX}/idris2-${IDRIS2_VERSION}/refc
